transformers/ns-replace-lazy-loader.ts

```diff
@@ -0,0 +1,301 @@
+        const ngModuleConfigObjectNode = ngModuleConfigNode as ts.ObjectLiteralExpression;
+        const matchingProperties: ts.ObjectLiteralElement[] = getObjectPropertyMatches(ngModuleConfigObjectNode, sourceFile, targetPropertyName);
+        if (!matchingProperties) {
+            // invalid object
+            return null;
+        }
+
+        if (matchingProperties.length == 0) {
+            if (ngModuleConfigObjectNode.properties.length == 0) {
+                // empty object @NgModule({ })
+                return null;
+            }
+
+            // the target field is missing, we will insert it @NgModule({ otherProps })
+            const lastConfigObjPropertyNode = ngModuleConfigObjectNode.properties[ngModuleConfigObjectNode.properties.length - 1];
+            const newTargetPropertyNode = ts.createIdentifier(`${targetPropertyName}: [${newPropertyValue}]`);
+
+            return [
+                new AddNodeOperation(sourceFile, lastConfigObjPropertyNode, undefined, newTargetPropertyNode),
+                new AddNodeOperation(sourceFile, lastImport, undefined, ngLazyLoaderNode)];
+
+        }
+
+        // the target property is found
+        const targetPropertyNode = matchingProperties[0] as ts.PropertyAssignment;
+        if (targetPropertyNode.initializer.kind !== ts.SyntaxKind.ArrayLiteralExpression) {
+            // not an array
+            return null;
+        }
+
+        const targetPropertyValuesNode = targetPropertyNode.initializer as ts.ArrayLiteralExpression;
+        const targetPropertyValues = targetPropertyValuesNode.elements;
+        if (targetPropertyValues.length > 0) {
+            // @NgModule({ targetProperty: [ someValues ] })
+            const targetPropertyValuesStrings = targetPropertyValues.map(node => node.getText());
+            const wholeWordPropValueRegex = new RegExp("\\b" + newPropertyValueMatch + "\\b");
+            if (targetPropertyValuesStrings.some(((value) => wholeWordPropValueRegex.test(value)))) {
+                // already registered
+                return null;
+            }
+
+            const lastPropertyValueNode = targetPropertyValues[targetPropertyValues.length - 1];
+            const newPropertyValueNode = ts.createIdentifier(`${newPropertyValue}`);
+
+            return [new AddNodeOperation(sourceFile, lastPropertyValueNode, undefined, newPropertyValueNode),
+            new AddNodeOperation(sourceFile, lastImport, undefined, ngLazyLoaderNode)];
+        } else {
+            // empty array @NgModule({ targetProperty: [ ] })
+            const newTargetPropertyValuesNode = ts.createIdentifier(`[${newPropertyValue}]`);
+
+            return [new ReplaceNodeOperation(sourceFile, targetPropertyValuesNode, newTargetPropertyValuesNode),
+            new AddNodeOperation(sourceFile, lastImport, undefined, ngLazyLoaderNode)];
+        }
+    }
+}
+
+function getDecoratorMetadata(source: ts.SourceFile, identifier: string,
```

Comment:
Should this be in the `ast-utils`?

Reply:
its depending on _angularImportsFromNode

Reply:
its here because its depending on collectDeepNodes from @ngtools/webpack/src/transformers and its more transformer specific logic